From 8f55155a7c09b5d6c81c7e08002ccd45824e9bda Mon Sep 17 00:00:00 2001
From: Li Huafei <lihuafei1@huawei.com>
Date: Thu, 14 Oct 2021 17:53:28 +0800
Subject: [PATCH] perf annotate: Add error log in symbol__annotate()
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: 8f55155a7c09b5d6c81c7e08002ccd45824e9bda


maillist inclusion
category: bugfix
bugzilla: 175624 https://gitee.com/openeuler/kernel/issues/I4DDEL

Reference: https://lkml.org/lkml/2021/7/26/406

--------------------------------

When users use the perf annotate feature on unsupported machines, error
logs should be printed for user feedback.

Signed-off-by: Li Huafei <lihuafei1@huawei.com>
Reviewed-by: James Clark <james.clark@arm.com>
Signed-off-by: Li Huafei <lihuafei1@huawei.com>
Reviewed-by: Kuohai Xu <xukuohai@huawei.com>
Signed-off-by: Chen Jun <chenjun102@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 tools/perf/util/annotate.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/tools/perf/util/annotate.c b/tools/perf/util/annotate.c
index 308189454788..4aaaf23b4878 100644
--- a/tools/perf/util/annotate.c
+++ b/tools/perf/util/annotate.c
@@ -2178,8 +2178,10 @@ int symbol__annotate(struct map_symbol *ms, struct evsel *evsel,
 		return errno;
 
 	args.arch = arch = arch__find(arch_name);
-	if (arch == NULL)
+	if (arch == NULL) {
+		pr_err("%s: unsupported arch %s\n", __func__, arch_name);
 		return ENOTSUP;
+	}
 
 	if (parch)
 		*parch = arch;
-- 
2.26.2

